srecord/arglex.cc

```diff
@@ -160,14 +160,14 @@ srecord::arglex::compare(const char *formal, const char *actual) -> bool
     for (;;)
     {
         unsigned char ac = *actual++;
-        if (isupper(ac)) {
+        if (isupper(ac) != 0) {
```

Comment:
The code is far more readable as-is. I'm ok with the implicit bool of a function called `isupper()`. Is there some kind of value in making this change?

I guess the plus side is that we also check the other cases where it makes less sense and should be corrected like above. Would be great if this could be done selectively.

Reply:
For a code checker it's hard to do selective checks. Even if you can configure it, you have to do the configuration for every code checker you use.

I do not like the C library for using int everywhere. `tolower` also returns int, although it's just a character.
I don't see C++ equivalent with bool. What do think about a long term solution to make a thin utility wrapper class for character handling:
~~~C++
  srecord:character ac = *actual++;
  if (ac.isupper())
~~~
and also add e.g. a `is_end_of_string()` method for `ac == 0U`

Reply:
Agreed. Some analysers have a finer grain set of tests (from memory, PVS Studio let's you parameterise some of them by type). But I completely take your point and I think I should just change my original answer as the majority of cases found aren't like what I described. 

Let's just go with your original proposal of testing to exclude implicit bool conversion and make the changes proposed.